tests/integration/cairo1/misc/gas_test/test.cairo

```diff
@@ -2,7 +2,7 @@ use array::ArrayTrait;
 
 
 fn fib(a: felt252, b: felt252, n: felt252) -> felt252 {
-    match gas::withdraw_gas() {
+    match gas::withdraw_gas_all(get_builtin_costs()) {
```

Comment:
could you please explain this change a bit?

Reply:
This shouldn't change anything in the behavior (I think/hope), but as per #1767 description, the syntax I introduced didn't work previously.

Perhaps we can have separate tests for both `withdraw_gas` and `withdraw_gas_all`?

Reply:
> Perhaps we can have separate tests for both `withdraw_gas` and `withdraw_gas_all`? 

That wouldn't be a bad idea.

Reply:
> Perhaps we can have separate tests for both withdraw_gas and withdraw_gas_all?
I would include them here

Reply:
Updated tests

Reply:
My description was probably a bit unclear, but I didn't mean that using `withdraw_gas` in the tests made using quaireaux fail. It was rather an observation - `withdraw_gas` worked with protostar, but quaireaux used `withdraw_gas_all(get_builtin_costs())` which made protostar fail.